src/python_testing/TC_SC_3_6.py

```diff
@@ -0,0 +1,226 @@
+        logging.info("Pre-conditions: writing initial value of NodeLabel, so that we can control for change of attribute detection")
+        await client_list[0].WriteAttribute(self.dut_node_id, [(0, Clusters.Basic.Attributes.NodeLabel(value=BEFORE_LABEL))])
+
+        # Subscribe with all clients to NodeLabel attribute
+        subscriptions = []
+        sub_handlers = []
+        resub_catchers = []
+
+        logging.info("Step 1 (first part): Establish subscription with all 15 clients")
+        for client in client_list:
+            logging.info("Establishing subscription from controller node %s" % client.name)
+            sub = await client.ReadAttribute(nodeid=self.dut_node_id, attributes=[(0, Clusters.Basic.Attributes.NodeLabel)],
+                                             reportInterval=(min_report_interval_sec, max_report_interval_sec), keepSubscriptions=False)
+            subscriptions.append(sub)
+
+            attribute_handler = AttributeChangeAccumulator(name=client.name)
+            sub.SetAttributeUpdateCallback(attribute_handler)
+            sub_handlers.append(attribute_handler)
+
+            resub_catcher = ResubscriptionCatcher(name=client.name)
+            sub.SetResubscriptionAttemptedCallback(resub_catcher, isAsync=True)
+            resub_catchers.append(resub_catcher)
+
+            if sub_liveness_override_ms is not None:
+                logging.warning("Overriding subscription liveness to check %dms! NOT FOR CERTIFICATION!" % sub_liveness_override_ms)
+                sub.OverrideLivenessTimeoutMs(sub_liveness_override_ms)
+
+        asserts.assert_equal(len(subscriptions), num_fabrics_to_commission *
+                             num_controllers_per_fabric, "Must have the right number of subscriptions")
+
+        # Trigger a change on NodeLabel
+        logging.info("Step 1 (second part): Change attribute with one client, and validate all clients observe the change on same session")
+        await asyncio.sleep(1)
+        await client_list[0].WriteAttribute(self.dut_node_id, [(0, Clusters.Basic.Attributes.NodeLabel(value=AFTER_LABEL))])
+
+        # Await a stabilization delay in increments to let the coroutines run
+        start_time = time.time()
+        while (time.time() - start_time) < stabilization_delay_sec:
+            await asyncio.sleep(0.05)
+
+        # After stabilization, validate no resubscriptions and all nodes have seen an update
+
+        logging.info("Validation of results")
+        # First check: all subs seeing update
+        failed = False
+        for handler in sub_handlers:
+            data_update_count = 0
+            for item in handler.all_data:
+                if item['value'] == AFTER_LABEL:
+                    data_update_count += 1
+
+            if data_update_count == 0:
+                logging.error("Client %s did not see subscription update" % handler.name)
+                failed = True
+            elif data_update_count > 1:
```

Comment:
This is brittle IMHO. Attributes are not meant to preserve edges, and should always be treated as level-based. This may be fine for this use-case, but it sets a precedence for others writing tests to view this in the wrong lens, and will lead to incorrect/unstable results.

Reply:
Again, the goal here is specifically to make sure that the subs did not die and that all communication worked/occurred. Without being able to check that, we can't check the overall goal of TC-SC-3.6, which is that you support N_FABRICS * N_CASE_SESSION_PER_FABRIC concurrent clients. Maybe checking no resubscriptions occurred after triggering the report would work

Reply:
Fixed